### ISSUE TITLE
barchart examples page has error in IE11 and Safari

### ISSUE THREAD

**Issue by jeanlange** [esnet/react-timeseries-charts#265]
TLDR; [example bar chart](http://software.es.net/react-timeseries-charts/#/example/barchart) page doesn't load in Safari.

----------------

I copied the [example bar chart](http://software.es.net/react-timeseries-charts/#/example/barchart) into my application to test, and it works fine in Chrome and FF; but when I open it in IE11, I get a JS error:
`Unhandled promise rejection TypeError: Unable to get property ‘begin’ of undefined or null resource`

I get (I think) the same error in Safari, but with a little more information about where it's coming from:
`TypeError: undefined is not an object (evaluating 'this.timerange().begin')`

I have tried polyfilling to solve my problem, but although I can see all of `core-js` being included, the error still occurs.

So then I thought maybe something was wrong with my integration, so I tried opening that example page in Safari, but got another error: `SyntaxError: Invalid character '\u2160'`

So there's:
- definitely a problem with looking at example pages with Safari (and maybe IE11)
- maybe a problem with (bar?) charts in Safari/IE11 in general.

I'm reporting this as the first issue because fixing that would let me see if the second issue is true.

**Comment by sartaj10** [comment]
Thanks for raising the issue @jeanlange 
Similar issue #201 has also been raised in the past. However, we've still not been able to identify and rectify the issue.

**Comment by jeanlange** [comment]
I see that the documentation site not working in Safari was already reported in #201 . It's marked as fixed and merged... but I'm still seeing the problem. Is there a release I need to wait for?

**Comment by sartaj10** [comment]
Nope, I don't know why it shows that over there. That issue is still open.

**Comment by jeanlange** [comment]
Ah, I see. PR #226 says it addresses it, and _that's_ merged. I do see that the issue is still open.

**Comment by sartaj10** [comment]
I believe when I made the PR a while ago, I mentioned it in the comments as a checkbox to fix it. However, I couldn't find a fix for it and thus removed it from the PR comments. 
Somehow it still shows that issue as part of the PR that was merged...

**Comment by hankthewhale** [comment]
@sartaj10 should the readme be updating with this info stating the current browser limitations?

**Comment by jeanlange** [comment]
Not sure if this is helpful help, but - I see a similar error being [caused by a minifier in webpack](https://github.com/webpack/webpack/issues/4510). You're not using that minifier, but maybe something to try?

Something they did over there to fix it was [set the charset encoding correctly](https://github.com/webpack/webpack/issues/4510#issuecomment-288078011).

**Comment by sartaj10** [comment]
@hankthewhale Sounds good. I'll do that

@jeanlange - Thanks for this. Looks like a fix was setting the encoding correctly. However, it's already set in our html file. I'll try to see more into this.
If you can figure out a fix for this, that'll be great as well.

**Comment by jeanlange** [comment]
I looked at this some more and found out which part was causing the error. Thanks for your codesandbox.io example, that was super helpful.

It looks like the time format the example data was in was not understandable to the library. Here's an [updated codesandbox.io example](https://codesandbox.io/s/oj61vk3nzy) that works on safari and (I assume but have not checked) IE. All I did was change datetime values in data.js from `"2017-01-24 00:00"` to `"2017-01-24T00:00"`

This isn't exactly a fix for this bug; it doesn't help the examples page to load, and it doesn't fix the examples in general. But maybe it'll give you a way forward.

**Comment by sartaj10** [comment]
@jeanlange 
Thanks for this! Upon changing this in the code, I was able to load the examples and the website perfectly locally in Safari. 

![screen shot 2018-05-15 at 9 05 31 am](https://user-images.githubusercontent.com/5708022/40069088-28039e3e-581f-11e8-887b-94fb4d7ed4f7.png)

**Comment by sartaj10** [comment]
Should be fixed in the upcoming release (v0.15.x)

**Comment by jeanlange** [comment]
Unfortunately, although the base issue I was having with the library was fixed and I can use it in my application in safari/ie11, the documentation website still doesn't load in Safari - still the same error,  `SyntaxError: Invalid character '\u2160'`

The codesandbox example is having trouble loading in any browser I attempt to open it in, not sure why (although the updated one that I provided above is loading for me just fine)

Seems to me like #201 should probably be reopened and this one should probably be retitled (just did that)

**Comment by sartaj10** [comment]
Yeah, this website loads fine in dev but the production version doesn't seem to.

Regarding the codesandbox example, there has been some issue with npm today and that's why it's having trouble downloading the updated version of the charts code.
